ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/AllocatedParties.scala

```diff
@@ -6,16 +6,20 @@ package com.daml.ledger.api.benchtool.submission
 import com.daml.ledger.client.binding.Primitive
 
 case class AllocatedParties(
-    signatory: Primitive.Party,
+    signatoryO: Option[Primitive.Party],
     observers: List[Primitive.Party],
     divulgees: List[Primitive.Party],
     extraSubmitters: List[Primitive.Party],
     observerPartySetO: Option[AllocatedPartySet],
 ) {
   val allAllocatedParties: List[Primitive.Party] =
-    List(signatory) ++ observers ++ divulgees ++ extraSubmitters ++ observerPartySetO.fold(
+    signatoryO.toList ++ observers ++ divulgees ++ extraSubmitters ++ observerPartySetO.fold(
       List.empty[Primitive.Party]
     )(_.parties)
+
+  /** NOTE: This is safe to call only you're sure signatory exists
+    */
+  def signatory: Primitive.Party = signatoryO.get
```

Comment:
NOTE: This is being addressed on a preceding PR now: https://github.com/digital-asset/daml/pull/14244